TourMate/TourMate/Frontend/Plan/Views/EditPlanViews/EditActivityView.swift

```diff
@@ -11,99 +11,74 @@ struct EditActivityView: View {
-                                venue: venue,
-                                phone: phone,
-                                website: website)
-        return activity
-    }
+    @StateObject var viewModel: EditPlanViewModel<Activity>
 
     var body: some View {
         NavigationView {
             Group {
                 if viewModel.hasError {
                     Text("Error occured")
                 } else {
-                    Form {
-                        Toggle("Confirmed?", isOn: $isConfirmed)
-                        TextField("Event Name", text: $eventName)
-                        DatePicker("Start Date",
-                                   selection: $startDate,
-                                   displayedComponents: [.date, .hourAndMinute])
-                        DatePicker("End Date",
-                                   selection: $endDate,
-                                   displayedComponents: [.date, .hourAndMinute])
-                        TextField("Venue", text: $venue)
-                        TextField("Address", text: $address)
-                        TextField("Phone", text: $phone)
-                        TextField("Website", text: $website)
-                    }
-                    .navigationTitle("Edit Activity")
-                    .navigationBarTitleDisplayMode(.inline)
-                    .toolbar {
-                        ToolbarItem(placement: .confirmationAction) {
-                            Button("Done") {
-                                Task {
-                                    await viewModel.updatePlan(plan: createUpdatedActivity())
-                                    dismiss()
-                                }
-                            }
-                            .disabled(viewModel.isLoading || viewModel.hasError)
+                    VStack {
+                        if !viewModel.canEditPlan {
+                            Text("Start date must be before end date")
+                                .font(.caption)
+                                .foregroundColor(.red)
                         }
-                        ToolbarItem(placement: .cancellationAction) {
-                            Button("Cancel", role: .destructive) {
-                                dismiss()
-                            }
-                            .disabled(viewModel.isLoading)
+                        Form {
+                            Toggle("Confirmed?", isOn: Binding<Bool>(
+                                get: { viewModel.plan.status == PlanStatus.confirmed },
+                                set: { select in
+                                    if select {
+                                        viewModel.plan.status = PlanStatus.confirmed
+                                    } else {
+                                        viewModel.plan.status = PlanStatus.proposed
+                                    }
+                                })
+                            )
```

Comment:
I think we should abstract the common logic out of the views into individual components, since they are duplicated across the 10 views we have

Reply:
Agreed, but should we can separate it into another PR? If not the changes might get larger and larger

Reply:
I feel like the changes are large because we have a lot of duplication right now. So every time we update plans the PR gets really big. But I'm also ok with merging first